### PR TITLE
Improve sprintf/printf for short format strings

### PR DESCRIPTION
**Pull request by forki**

### CODE REVIEW

**Comment by forki** [comment]
I used https://github.com/fsprojects/FSCBenchmark and it shows very good success on stuff like:

    sprintf "%A" myObject

image

**Comment by forki** [comment]
image

**Comment by forki** [comment]
ready for review

File: src/fsharp/FSharp.Core/printf.fs [hunk 1]
```diff
@@ -1087,14 +1272,47 @@ module internal PrintfImpl =
...
+#endif
+                let mi = mi.MakeGenericMethod(argTypes)
+                let args = Array.sub args 0 (argsCount - 1)
+                optimizedArgCount <- optimizedArgCount + 1
+                mi.Invoke(null, args)                
+            else
+                let mi = typeof<Specializations<'S, 'Re, 'Res>>.GetMethod("Final" + count, NonPublicStatics)
+#if DEBUG
+                verifyMethodInfoWasTaken mi
+#else
```
Comment (dsyme) [inline src/fsharp/FSharp.Core/printf.fs]: remove empty else

Comment (forki) [reply src/fsharp/FSharp.Core/printf.fs]: Empty Else?

Comment (forki) [reply src/fsharp/FSharp.Core/printf.fs]: ah I see

**Comment by KevinRansom** [comment]
@forki 

Thank you

**Pull request merged**